### PR TITLE
Clear stored username/token from config when relevant checkboxes are unticked on login panel

### LINKED ISSUE #27272: Remember username still remembers your username when disabled.
### Type

Cosmetic

### Bug description

Logging out of Lazer and disabling the "Remember Username" option after it was previously on will cause your username to still appear in the text field on start-up.

### Screenshots or videos

https://github.com/ppy/osu/assets/37195727/58f805bb-7c8f-4e61-8751-fca0d6f9685f

### Version

2024.219.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14346929/compressed-logs.zip)

### PR DESCRIPTION
**Pull request by BoudewijnPopkema**
Resolves #27272 by clearing the saved Username when "Remember Username" is unchecked.

### CODE REVIEW

**Comment by bdach** [comment]
I've basically rewritten this to use a more correct implementation with test coverage.

Will wait for a second review from someone else due to the above.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**
